immuni_common/helpers/sanic.py

```diff
@@ -178,3 +184,52 @@ def _validate_json_content_type(request: Request) -> None:
         return _wrapper
 
     return _decorator
+
+
+async def wait_configured_time() -> None:
+    """
+    Wait for the configured time.
+    This is usually useful to make dummy requests last a similar amount of time when compared to the
+    real ones, or slow down potential brute force attacks.
+    """
+    await asyncio.sleep(
+        random.normalvariate(
+            config.DUMMY_REQUEST_TIMEOUT_MILLIS, config.DUMMY_REQUEST_TIMEOUT_SIGMA
+        )
+        / 1000.0
+    )
+
+
+def handle_dummy_requests(
+    responses: List[WeightedPair],
+) -> Callable[
+    [Callable[..., Coroutine[Any, Any, HTTPResponse]]],
+    Callable[..., Coroutine[Any, Any, HTTPResponse]],
+]:
+    """
+    Decorator that allows handling dummy requests.
+    :param responses: A list of WeightedPair, where the payload is the response
+     to be returned with the given weight. The decorator will pick a random
+     response based on the given weights.
+    :return: The decorated function.
+    """
+
+    def _decorator(
+        f: Callable[..., Coroutine[Any, Any, HTTPResponse]]
+    ) -> Callable[..., Coroutine[Any, Any, HTTPResponse]]:
+        @validate(
+            location=Location.HEADERS,
+            is_dummy=IntegerBoolField(
+                required=True, allow_strings=True, data_key=HeaderImmuniDummyData.DATA_KEY,
+            ),
+        )
+        @doc.consumes(HeaderImmuniDummyData(), location="header", required=True)
+        async def _wrapper(*args: Any, is_dummy: bool, **kwargs: Any) -> HTTPResponse:
+            if is_dummy:
+                await wait_configured_time()
+                return weighted_random(responses)
```

Comment:
I'm not super-convinced on the `weighted_random` name. 🤔
I'm thinking about alternatives but cannot find a good one. Ideally, you should inform the caller that the method returns a single response, so maybe `weighted_random_response` would do already? (maybe even without `random`?)